### PR TITLE
[RHELC-1488]  Remove packages left after the TF artifact installation 

### PR DESCRIPTION
**Pull request by kokesak**
After the artifact installation done by TF there are some packages left on the system.
They are considered as a third party packages so we want to remove them from the system as well
as the corresponding repositories.

Also fix single-yum-transaction-validation int tests

Jira Issues: [RHELC-1488](https://issues.redhat.com/browse/RHELC-1488)

Checklist

- [x] PR has been tested manually in a VM (either author or reviewer)
- [x] Jira issue has been made public if possible
- [x] `[RHELC-]` is part of the PR title 
- [x] GitHub label has been added to help with Release notes 
- [x] PR title explains the change from the user's point of view
- [x] Code and tests are documented properly
- [x] The commits are squashed to as few commits as possible (without losing data) 
- [ ] When merged: Jira issue has been updated to `Release Pending` if relevant

### CODE REVIEW

File: tests/ansible_collections/roles/remove-rcmtools/tasks/clean_up_el7.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+- name: Remove rcmtools packages installed on the system
+  # We cannot use yum remove as it also removes dependency packages (like even c2r) that we need to keep installed
+  ansible.builtin.shell: yum list installed | grep @rcmtools | awk '{print $1}' | xargs rpm -e --nodeps
+- name: Remove rcmtools repo from the system
+  ansible.builtin.shell: rm -f /etc/yum.repos.d/rcmtools.repo
```
Comment (danmyway) [inline tests/ansible_collections/roles/remove-rcmtools/tasks/clean_up_el7.yml:6]:
```suggestion
  ansible.builtin.yum_repository:
    file: rcmtools
    name: rcmtools
    state: absent

- name: Clean yum metadata
  ansible.builtin.command: yum clean metadata
```
Can we rather go with the [built-in yum_repository module](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/yum_repository_module.html) ?
Ideally, clean the metadata as well.

File: tests/ansible_collections/roles/remove-rcmtools/tasks/clean_up_el8.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+- name: Remove rcmtools packages installed on the system
+  ansible.builtin.shell: dnf list --installed | grep @rcmtools | awk '{print $1}' | xargs dnf remove -y
+- name: Remove rcmtools repo from the system
+  ansible.builtin.shell: rm -f /etc/yum.repos.d/rcmtools.repo
```
Comment (danmyway) [inline tests/ansible_collections/roles/remove-rcmtools/tasks/clean_up_el8.yml:5]:
```suggestion
  ansible.builtin.yum_repository:
    file: rcmtools
    name: rcmtools
    state: absent

- name: Clean yum metadata
  ansible.builtin.command: yum clean metadata
```
Same as in clean_up_el8.yml

**Review by danmyway — CHANGES_REQUESTED** [review]
Only small ask to use the builtin modules instead of plain shell in some places.

File: tests/ansible_collections/roles/remove-rcmtools/tasks/clean_up_el7.yml (hunk 1 above)
Comment (kokesak) [reply tests/ansible_collections/roles/remove-rcmtools/tasks/clean_up_el7.yml:6]: ok, I'll change it

File: tests/ansible_collections/roles/remove-rcmtools/tasks/clean_up_el8.yml (hunk 2 above)
Comment (kokesak) [reply tests/ansible_collections/roles/remove-rcmtools/tasks/clean_up_el8.yml]: Done

**Review by danmyway — APPROVED** [review]
Looks good to me, failing tests are unrelated.

**Pull request merged**